From 5dd9e7adddbc080a92444880bf0ba3e60c749729 Mon Sep 17 00:00:00 2001
From: Adrian Reber <areber@redhat.com>
Date: Sun, 27 Apr 2025 12:28:18 +0200
Subject: [PATCH] Fix errors when building an RPM

Building an RPM for OpenHPC of pnetcdf fails with:

+ /usr/lib/rpm/check-buildroot
/home/ohpc/rpmbuild/BUILDROOT/pnetcdf-gnu14-mvapich2-ohpc-1.14.0-19999.ci.ohpc.x86_64/opt/ohpc/pub/libs/gnu14/mvapich2/pnetcdf/1.14.0/bin/pnetcdf-config:prefix=/home/ohpc/rpmbuild/BUILDROOT/pnetcdf-gnu14-mvapich2-ohpc-1.14.0-19999.ci.ohpc.x86_64/opt/ohpc/pub/libs/gnu14/mvapich2/pnetcdf/1.14.0
/home/ohpc/rpmbuild/BUILDROOT/pnetcdf-gnu14-mvapich2-ohpc-1.14.0-19999.ci.ohpc.x86_64/opt/ohpc/pub/libs/gnu14/mvapich2/pnetcdf/1.14.0/bin/pnetcdf-config:exec_prefix=/home/ohpc/rpmbuild/BUILDROOT/pnetcdf-gnu14-mvapich2-ohpc-1.14.0-19999.ci.ohpc.x86_64/opt/ohpc/pub/libs/gnu14/mvapich2/pnetcdf/1.14.0
/home/ohpc/rpmbuild/BUILDROOT/pnetcdf-gnu14-mvapich2-ohpc-1.14.0-19999.ci.ohpc.x86_64/opt/ohpc/pub/libs/gnu14/mvapich2/pnetcdf/1.14.0/lib/pkgconfig/pnetcdf.pc:prefix=/home/ohpc/rpmbuild/BUILDROOT/pnetcdf-gnu14-mvapich2-ohpc-1.14.0-19999.ci.ohpc.x86_64/opt/ohpc/pub/libs/gnu14/mvapich2/pnetcdf/1.14.0
/home/ohpc/rpmbuild/BUILDROOT/pnetcdf-gnu14-mvapich2-ohpc-1.14.0-19999.ci.ohpc.x86_64/opt/ohpc/pub/libs/gnu14/mvapich2/pnetcdf/1.14.0/lib/pkgconfig/pnetcdf.pc:exec_prefix=/home/ohpc/rpmbuild/BUILDROOT/pnetcdf-gnu14-mvapich2-ohpc-1.14.0-19999.ci.ohpc.x86_64/opt/ohpc/pub/libs/gnu14/mvapich2/pnetcdf/1.14.0
Found '/home/ohpc/rpmbuild/BUILDROOT/pnetcdf-gnu14-mvapich2-ohpc-1.14.0-19999.ci.ohpc.x86_64' in installed files; aborting

The reason is that the variable DESTDIR is added to the path of the
binary location. This is, however, wrong. DESTDIR is the directory where
the files are installed but not where they are location in the final
RPM.

Keeping DESTDIR for the file location is correct but not for the path
where the files are located.

With this patch applied, it is possible to create an RPM without errors
like above.

Signed-off-by: Adrian Reber <areber@redhat.com>
---
 src/packaging/Makefile.am | 2 +-
 src/utils/Makefile.am     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/packaging/Makefile.am b/src/packaging/Makefile.am
index 80bd45bf..3431f856 100644
--- a/src/packaging/Makefile.am
+++ b/src/packaging/Makefile.am
@@ -12,6 +12,6 @@ pkgconfig_DATA = pnetcdf.pc
 # GNU autoconf allows DESTDIR variable when running 'make install' which will
 # prepend it before all installation names.
 install-data-hook:
-	$(SED_I) -e 's|INSTALL_PREFIX|$(DESTDIR)$(prefix)|g ; s|INSTALL_EXEC_PREFIX|$(DESTDIR)$(exec_prefix)|g' $(DESTDIR)$(libdir)/pkgconfig/pnetcdf.pc
+	$(SED_I) -e 's|INSTALL_PREFIX|$(prefix)|g ; s|INSTALL_EXEC_PREFIX|$(exec_prefix)|g' $(DESTDIR)$(libdir)/pkgconfig/pnetcdf.pc
 
 
diff --git a/src/utils/Makefile.am b/src/utils/Makefile.am
index 3e3c905b..a0b931fe 100644
--- a/src/utils/Makefile.am
+++ b/src/utils/Makefile.am
@@ -36,6 +36,6 @@ dist-hook:
 # GNU autoconf allows DESTDIR variable when running 'make install' which will
 # prepend it before all installation names.
 install-exec-hook:
-	$(SED_I) -e 's|INSTALL_PREFIX|$(DESTDIR)$(prefix)|g ; s|INSTALL_EXEC_PREFIX|$(DESTDIR)$(exec_prefix)|g' $(DESTDIR)$(bindir)/pnetcdf-config
+	$(SED_I) -e 's|INSTALL_PREFIX|$(prefix)|g ; s|INSTALL_EXEC_PREFIX|$(exec_prefix)|g' $(DESTDIR)$(bindir)/pnetcdf-config
 	chmod +x $(DESTDIR)$(bindir)/pnetcdf-config
 
-- 
2.49.0

